components/TextField.tsx

```diff
@@ -0,0 +1,77 @@
+import { FC, useState, useEffect, useRef, ReactNode } from 'react'
+
+import {
+  Box,
+  Flex,
+  FlexProps,
+  Input,
+  InputProps as IProps,
+  useMultiStyleConfig,
+} from '@chakra-ui/react'
+
+interface TextFieldProps extends FlexProps {
+  label: string
+  value?: string
+  InputProps?: IProps
+  LeftAddons?: ReactNode
+  RightAddons?: ReactNode
+}
+
+const TextField: FC<TextFieldProps> = ({
+  label,
+  value = '',
+  InputProps = {},
+  LeftAddons = null,
+  RightAddons = null,
+  ...props
+}) => {
+  const [val, setVal] = useState<string>(value)
+  const inputRef = useRef<HTMLInputElement>(null)
+  const styles = useMultiStyleConfig('TextField', props)
+
+  useEffect(() => {
+    setVal(value)
+  }, [value])
+
+  return (
+    <Flex
+      id="TextField"
+      tabIndex={0}
+      onFocus={() => inputRef?.current?.focus()}
+      {...props}
+      sx={{
+        ...styles?.container,
+        ...props.sx,
+      }}
+      role="group"
+      justifyContent="center"
+      alignItems="center"
+    >
+      {LeftAddons}
+      <Flex flexDirection="column" justifyContent="center" w="100%">
+        <Box sx={styles?.label}>{label}</Box>
+        <Box
+          sx={styles?.inputWrapper}
+          display={!!val ? 'block' : 'none'}
+          _groupFocus={{ display: 'block' }}
+          _groupFocusVisible={{ display: 'block' }}
+          _groupFocusWithin={{ display: 'block' }}
```

Comment:
Minor nit, but could `{display: 'block'}` be defined as a standalone variable and re-used here and the previous few lines?